### PR TITLE
Keeping the vulnerability detector DB synced cleaning all the rejected CVE references 

### LINKED ISSUE #20388: Properly update the vulnerability detector internal DBs in case a previously PUBLISHED document becomes REJECTED
## Description

All the new elements that the `contentManager` publishes to the `databaseFeedManager` are processeced in `storeModel` to update the `descriptions`, `remediations`, `candidates` and `translations` internal DBs.

The case when a CVE 5.0 document is first informed as `PUBLISHED` but then in an update changes to `REJECTED` has to be properly handled to avoid leaving invalid information in the mentioned DBs.

## DoD

- [x] Share a comment with the analysis of the situation and the possible solution
- [x] Update the required orchestration to handle the described situation
- [x] Perform manual tests, changing a CVE from one state to another. Include other tests if required
- [x] Add UT
- [x] CI Pass

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #20388|

## Description

This PR creates a mechanism to keep the candidates, descriptions and remediations columns synced in case a CVE changes or becomes rejected.

Also, a new parameter to the `rocksDB` test tool was added; the DB path for the offset now is set in case an offline feed is configured; and the `databaseFeedManager` test tool was updated.

## Tests

The UT added verify the behavior of the new methods added.

The `databaseFeedManager` test tool was used to confirm that a rejected entry removes the corresponding entries

```
./database_feed_manager_testtool -f offsets -c config.json -i file_offset1.json
```

```
./rocks_db_query_testtool -d queue/vd/feed/ -c cve_package
CVE-1999-1234_bash ==> bash_CVE-1999-1234
CVE-1999-1234_bash b ==> bash b_CVE-1999-1234
```

```
 ./rocks_db_query_testtool -d queue/vd/feed/ -c nvd -f vulnerabilityCandidate.fbs
bash b_CVE-1999-1234 ==> {
  "candidates": [
    {
      "cveId": "CVE-1999-1234",
      "defaultStatus": "unaffected",
      "platforms": [

      ],
      "versions": [
        {
          "version": "2.05"
        }
      ]
    }
  ]
}

 ./rocks_db_query_testtool -d queue/vd/feed/ -c debian -f vulnerabilityCandidate.fbs
bash_CVE-1999-1234 ==> {
  "candidates": [
    {
      "cveId": "CVE-1999-1234",
      "platforms": [
        "upstream"
      ],
      "versions": [
        {
          "version": "0",
          "lessThan": "98.0.4758.102",
          "versionType": "custom"
        }
      ]
    }
  ]
}

```

After this command, all columns are empty
```
./database_feed_manager_testtool -f offsets -c config.json -i file_offset2.json
```

The scanner test tool was used to confirm that the new column is being properly populated using a small offline file

The current online feed was completely parsed successfully

(Work in progress)

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by GabrielEValenzuela — COMMENTED** [review]
Hi @pereyra-m !

I tried to reproduce the test steps but when I try to run the testtool I get

```bash
==126510==ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
```

Maybe this could fix adding `address_sanitazer` as a linking library in the `CMakeLists.txt`

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM! The overall structure is clear, and the code is easy to read and understand.

### Readability:
The variable names are descriptive, and the comments provide helpful insights into the logic.

### Testing:
I've executed the provided tests, and I'm pleased to report that they produced the expected results.

<details><summary>Test report</summary>
#### Test requisites
- Run `scanner` testtool to fill the databases.

#### Test case: Delete entry
1 - Command execution

```bash
./src/build/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/rocks_db_query_testtool -d queue/vd/feed -c cve_package | grep "1999-1234"
```
Output
```text
CVE-1999-1234_bash ==> bash_CVE-1999-1234
CVE-1999-1234_bash b ==> bash b_CVE-1999-1234
```

1 - Delete offset
```bash
./src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/database_feed_manager_testtool -c src/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/config.json -f offsets -i src/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/file_offset2.json
```
Output
```text

```

#### Test case: Update entry
1 - Command execution

```bash
./src/build/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/rocks_db_query_testtool -d queue/vd/feed -c cve_package | grep "1999-1234"
```
Output
```text
CVE-1999-1234_bash ==> bash_CVE-1999-1234
CVE-1999-1234_bash b ==> bash b_CVE-1999-1234
```

1 - Delete offset
```bash
./src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/database_feed_manager_testtool -c src/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/config.json -f offsets -i src/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/file_offset3.json
```
Output
```text
CVE-1999-1234_bash ==> bash_CVE-1999-1234
CVE-1999-1234_bash b ==> bash b_CVE-1999-1234
CVE-1999-1234_edge ==> edge_CVE-1999-1234
CVE-1999-1234_vim ==> vim_CVE-1999-123
```
</details>

Great job Mati!

**Review by sebasfalcone — DISMISSED** [review]
LGTM

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp [hunk 1]
```diff
@@ -32,19 +32,45 @@ class StoreModel final : public AbstractHandler<std::shared_ptr<EventContext>>
...
+                    UpdateCVEDescription::storeVulnerabilityDescription(cve5Entry, data->feedDatabase);
+                    UpdateCVECandidates::storeVulnerabilityCandidate(cve5Entry, data->feedDatabase);
+                }
+                else if ("REJECTED" == state)
+                {
+                    StoreRemediationsModel::removeRemediation(cve5Entry, data->feedDatabase);
+                    UpdateCVEDescription::removeVulnerabilityDescription(cve5Entry, data->feedDatabase);
+                }
+            }
+            else if ("create" == type)
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp:63]: Source code that utilizes if-elseif control statements must terminate with an else statement.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp [hunk 2]
```diff
@@ -32,19 +32,45 @@ class StoreModel final : public AbstractHandler<std::shared_ptr<EventContext>>
...
+            {
+                // We clean the candidates DBs to keep the data synced
+                UpdateCVECandidates::removeVulnerabilityCandidate(cve5Entry, data->feedDatabase);
+                if ("PUBLISHED" == state)
+                {
+                    StoreRemediationsModel::updateRemediation(cve5Entry, data->feedDatabase);
+                    UpdateCVEDescription::storeVulnerabilityDescription(cve5Entry, data->feedDatabase);
+                    UpdateCVECandidates::storeVulnerabilityCandidate(cve5Entry, data->feedDatabase);
+                }
+                else if ("REJECTED" == state)
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp:57]: Source code that utilizes if-elseif control statements must terminate with an else statement.

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp [hunk 3]
```diff
@@ -32,19 +32,45 @@ class StoreModel final : public AbstractHandler<std::shared_ptr<EventContext>>
...
-                UpdateCVEDescription::storeVulnerabilityDescription(cve5Entry, data->feedDatabase);
+                if ("PUBLISHED" == state)
+                {
+                    StoreRemediationsModel::updateRemediation(cve5Entry, data->feedDatabase);
+                    UpdateCVEDescription::storeVulnerabilityDescription(cve5Entry, data->feedDatabase);
+                    UpdateCVECandidates::storeVulnerabilityCandidate(cve5Entry, data->feedDatabase);
+                }
+                else if ("REJECTED" == state)
+                {
+                }
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp:73]: Done!

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp:63]: Done!

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/storeModel.hpp:57]: Done!

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**